lms/djangoapps/courseware/grades.py

```diff
@@ -476,7 +530,13 @@ def get_weighted_scores(student, course):
     return _progress_summary(student, course)
 
 
-def _progress_summary(student, course, course_structure=None):
+def _progress_summary(
+    student,
+    course,
+    course_structure=None,
```

Comment:
Suggestion: Instead of having to create optional parameters like this for efficiency, consider the following restructure of these functions.  
1. Introduce a new method called something like "prepare_grading_context" or "prep_for_grading".  This new method would then do all 3 prework, pre-caching, etc.   It returns a context to be passed into the dependent grading methods.
2. Update all callers (aren't that many) to "grade" and "progress_summary" to call the "prep" function and pass in the context to the "grade" and "progress_summary" functions.

Essentially, require that all callers pass in the context.  So you don't have to conditionally check for them and duplicate calls to create them, if not passed in.  (I believe there's a section in the Clean Code book that may have a similar example.)